features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/PrivacyOverrideManager.kt

```diff
@@ -0,0 +1,107 @@
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal
+
+import android.view.View
+import com.datadog.android.sessionreplay.ImagePrivacy
+import com.datadog.android.sessionreplay.PrivacyLevel
+import com.datadog.android.sessionreplay.TextAndInputPrivacy
+import com.datadog.android.sessionreplay.TouchPrivacy
+import java.lang.ref.WeakReference
+import java.util.concurrent.ConcurrentHashMap
+
+internal object PrivacyOverrideManager {
+    private val overridesMap = ConcurrentHashMap<WeakViewKey, PrivacyOverride>()
+
+    internal fun addPrivacyOverride(view: View, level: PrivacyLevel?) {
+        val key = WeakViewKey(view)
+        val existingPrivacy = overridesMap[key] ?: PrivacyOverride()
+
+        overridesMap[key] =
+            when (level) {
+                is ImagePrivacy -> existingPrivacy.copy(
+                    imagePrivacy = level
+                )
+
+                is TextAndInputPrivacy -> existingPrivacy.copy(
+                    textAndInputPrivacy = level
+                )
+
+                is TouchPrivacy -> existingPrivacy.copy(
+                    touchPrivacy = level
+                )
+
+                else -> return
+            }
+    }
+
+    internal fun removeTextAndInputPrivacyOverride(view: View) {
+        val key = WeakViewKey(view)
+        val existingPrivacy = overridesMap[key] ?: return
+
+        overridesMap[key] = existingPrivacy.copy(
+            textAndInputPrivacy = null
+        )
+    }
+
+    internal fun removeTouchPrivacyOverride(view: View) {
+        val key = WeakViewKey(view)
+        val existingPrivacy = overridesMap[key] ?: return
+
+        overridesMap[key] = existingPrivacy.copy(
+            touchPrivacy = null
+        )
+    }
+
+    internal fun removeImagePrivacyOverride(view: View) {
+        val key = WeakViewKey(view)
+        val existingPrivacy = overridesMap[key] ?: return
+
+        overridesMap[key] = existingPrivacy.copy(
+            imagePrivacy = null
+        )
+    }
+
+    internal fun addHiddenOverride(view: View) {
+        val key = WeakViewKey(view)
+        val existingPrivacy = overridesMap[key] ?: PrivacyOverride()
+
+        overridesMap[key] = existingPrivacy.copy(
+            hiddenPrivacy = true
+        )
+    }
+
+    internal fun removeHiddenOverride(view: View) {
+        val key = WeakViewKey(view)
+        val existingPrivacy = overridesMap[key] ?: return
+
+        overridesMap[key] = existingPrivacy.copy(
+            hiddenPrivacy = false
+        )
+    }
+
+    internal fun getPrivacyOverrides(view: View): PrivacyOverride? {
+        val key = WeakViewKey(view)
+        return overridesMap[key]
+    }
+
+    internal fun isHidden(view: View): Boolean {
+        val key = WeakViewKey(view)
+        return overridesMap[key]?.hiddenPrivacy == true
+    }
+
+    private class WeakViewKey(view: View) : WeakReference<View>(view) {
```

Comment:
blocking:

Not quite the best solution, because your View will still be in the memory even after the screen is destroyed until GC kicks in. Is there are other way to keep view identity? I think in the Session Replay we are relying solely on the hash code, which is enough, because once existing view is destroyed, the new one created will still have `ddSessionReplayOverride....` call made on it if needed.